core/src/test/scala/integration/kafka/api/RackAwareAutoTopicCreationTest.scala

```diff
@@ -0,0 +1,70 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.api
+
+import java.util.Properties
+
+import kafka.admin.{RackAwareMode, AdminUtils, RackAwareTest}
+import kafka.integration.KafkaServerTestHarness
+import kafka.server.KafkaConfig
+import kafka.utils.{ZkUtils, TestUtils}
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.junit.Assert._
+import org.junit.Test
+import org.scalatest.junit.JUnit3Suite
+import scala.collection.Map
+
+class RackAwareAutoTopicCreationTest extends KafkaServerTestHarness with RackAwareTest {
```

Comment:
We have unit tests for AdminUtils.assignReplicasToBrokers() that does not require starting a server. However, since the logic that governs the rack aware mode is separate from AdminUtils.assignReplicasToBrokers(), we need to have tests to make sure the right API is called from KafkaApi. The behavior we need to test is

- If all brokers have rack, rack aware assignment will be generated
- If some brokers have rack and some do not, it will be treated as none of the brokers have rack